File: java/src/main/java/deltix/luminary/DecoratorPropertyDef.java
package deltix.luminary;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public class DecoratorPropertyDef extends PropertyDef<DecoratorDef> implements Defaultable {
    private Literal defaultValue;

    public DecoratorPropertyDef(@NotNull DecoratorDef owner, @NotNull String name, @NotNull Type type,
                                @Nullable Literal defaultValue, @Nullable List<String> comments) {
        super(owner, name, type, comments);
        this.defaultValue = defaultValue;
        owner.getProperties().add(this);
    }

    public DecoratorPropertyDef(@NotNull DecoratorDef owner, @NotNull String name, @NotNull Type type,
                                @Nullable Literal defaultValue) {
        this(owner, name, type, defaultValue, null);
    }

    @Nullable
    @Override
    public Literal getDefault() {
        return defaultValue;
    }

    void setDefault(Literal defaultValue) {
        this.defaultValue = defaultValue;
    }
}

File: csharp/main/DecoratorPropertyDef.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Deltix.Luminary
{
	public sealed class DecoratorPropertyDef : PropertyDef<DecoratorDef>
	{
		public DecoratorPropertyDef([NotNull] DecoratorDef owner, [NotNull] String name, [NotNull] Type type, [CanBeNull] Literal @default, [CanBeNull, ItemNotNull] List<String> comments = null)
			: base(owner, name, type, comments)
		{
			Default = @default;

			FullName = $"{owner.FullName}.{Name}";

			owner.Properties.Add(this);
		}

		[CanBeNull]
		public Literal Default { get; internal set; }

		public override String FullName { get; }
	}
}
